cmd/tailscale/cli/netcheck.go

```diff
@@ -59,7 +60,14 @@ func runNetcheck(ctx context.Context, args []string) error {
 		fmt.Fprintln(os.Stderr, "# Warning: this JSON format is not yet considered a stable interface")
 	}
 
-	dm := derpmap.Prod()
+	ctx, cancel := context.WithTimeout(ctx, 500*time.Millisecond)
+	defer cancel()
+
+	dm, err := tailscale.CurrentDERPMap(ctx)
+	if err != nil || dm == nil {
+		// TODO should this log something?
+		dm = derpmap.Prod()
```

Comment:
I'd like to entirely remove the derpmap.Prod from the open source tree and fall back to fetching it over the network from login.tailscale.com instead.

We might need to add a new server handler for that first.